.github/workflows/gt4py-tox.yml

```diff
@@ -30,4 +30,4 @@ jobs:
         pyversion_no_dot="${{ matrix.python-version }}"
         pyversion_no_dot="${pyversion_no_dot/./}"
         pip install tox
-        tox -r -e py${pyversion_no_dot}-cpu
+        tox -r -e py${pyversion_no_dot}-dawn-cpu
```

Comment:
Does this include the other backends?

Reply:
yes, this is using a new config in `tox.ini` which adds dawn as a dependency, and thus dawn backends are also included in the tests

Reply:
Of course this means that dawn is being checkout and built as part of the venv creation, so the tests will take longer and maybe would be better to revert this change. Let's give it a try to see if it works and how long it takes, and we can decide later

Reply:
It double the time to 22 mins (before 11 mins). I'd keep it for now and we can check if it gets annoying. It is anyway faster than our typical jenkins tests...

Reply:
Do you know how to create a matrix where the "dawn" string appears or not?

Something like:  `tox -r -e py${pyversion_no_dot}{option}cpu ` where option is either `-dawn-` or just `-`

In this way it would launch both tasks in parallel (with dawn and without dawn) and then there would preliminary results ready in 11 mins

Reply:
I added something that seems to work, but not sure if this is the right way to do it. Could you take a look at it?

Reply:
Yes that's correct, though you don't have to go via a shell variable.

Reply:
But now you are still running the full thing in `dawn-`, right?

Reply:
Yes, both versions are running in parallel but you get the results of the no-dawn version earlier, so I thought it might be useful. I can remove it though, it was mostly an excuse to play with github actions setup

Reply:
If it's easy to decompose tests for different backends, we could run all of them in parallel in github actions. (Or just running dawn only tests in the current dawn- test would also speed up the whole thing.)